src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,64 @@
+package duke.parser;
+
+import duke.exception.IllegalTaskCommandException;
+
+import java.util.StringJoiner;
+
+public class Parser {
+    public static String getCommand(String userInput) {
+        String[] inputArray = userInput.split(" ");
+        String command = inputArray[0];
+        return command.toUpperCase();
+    }
+
+    public static String getSubstring(String userInput) throws IllegalTaskCommandException {
+        String cleanUserInput = userInput.strip(); // Removes any leading and trailing spaces
+        int spacePosition = cleanUserInput.indexOf(" ");
+        if (spacePosition < 0) { // Is single command

```

Comment:
Do I see a magic number here?

Reply:
indexOf would return a "-1" should the index of the item not be found. But I think i'll work directly with that. Thank you for the comment!